hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystem.java

```diff
@@ -610,12 +602,68 @@ public void testListStatusWithIntermediateDir() throws Exception {
     }
 
     FileStatus[] fileStatuses = fs.listStatus(parent);
-
     // the number of immediate children of root is 1
     Assert.assertEquals(1, fileStatuses.length);
     writeClient.deleteKey(keyArgs);
   }
 
+  @Test
+  public void testListStatusWithIntermediateDirWithECEnabled() throws Exception {
+    System.out.println("Executing");
+    String keyName = "object-dir/object-name1";
+    OmKeyArgs keyArgs = new OmKeyArgs.Builder()
+            .setVolumeName(volumeName)
+            .setBucketName(bucketName)
+            .setKeyName(keyName)
+            .setAcls(Collections.emptyList())
+            .setReplicationConfig(new ECReplicationConfig(3, 2))
+            .setLocationInfoList(new ArrayList<>())
+            .build();
+    OpenKeySession session = writeClient.openKey(keyArgs);
+
+    writeClient.commitKey(keyArgs, session.getId());
+
+
+
+
+
+
+
+    keyName = "object-dir/object-name2";
+    keyArgs = new OmKeyArgs.Builder()
+            .setVolumeName(volumeName)
+            .setBucketName(bucketName)
+            .setKeyName(keyName)
+            .setAcls(Collections.emptyList())
+            .setReplicationConfig(new ECReplicationConfig(3, 2))
+            .setLocationInfoList(new ArrayList<>())
+            .build();
+    session = writeClient.openKey(keyArgs);
+
+    writeClient.commitKey(keyArgs, session.getId());
+
+    Path parent = new Path("/");
+
+    // Wait until the filestatus is updated
+    if (!enabledFileSystemPaths) {
+      GenericTestUtils.waitFor(() -> {
+        try {
+          return fs.listStatus(parent).length != 0;
+        } catch (IOException e) {
+          LOG.error("listStatus() Failed", e);
+          Assert.fail("listStatus() Failed");
+          return false;
+        }
+      }, 1000, 120000);
+    }
```

Comment:
is this necessary? I am under the impression commitKey() is synchronous and listStatus() should reflect the change immediately?